File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.cpp
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        int n = nums1.size(), m = nums2.size();
        int k = n + m, midInd = k / 2;
        int mid = 0, preMid = 0, i = 0, j = 0, cnt = 0;

        while (i < n || j < m) {
            int val;
            if (i < n && (j >= m || nums1[i] < nums2[j])) {
                val = nums1[i++];
            } else {
                val = nums2[j++];
            }
            if (cnt == midInd - 1) preMid = val;
            if (cnt == midInd) {
                mid = val;
                break;
            }
            cnt++;
        }

        return (k % 2 == 0) ? (mid + preMid) / 2.0 : mid;
    }
};

File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int n=nums1.length;
        int m=nums2.length;

        int curr=-1, prev=-1;
        int cnt=0;
        int medianPointer=(n+m)/2;

        double ans=0;

        int l=0,r=0;

        while(cnt<=medianPointer){
            prev=curr;
            if(l<n && (r>=m || nums1[l]<=nums2[r])){
                curr=nums1[l];
                l++;
            }else{
                curr=nums2[r];
                r++;
            }
            cnt++;
        }

        if((n+m)%2==0){
            ans= (curr+prev)/2.0;
        }else{
            ans=curr;
        }
        return ans;
    }
}